src/main/java/com/epam/ta/reportportal/core/item/impl/UpdateTestItemHandlerImpl.java

```diff
@@ -173,8 +173,7 @@ public OperationCompletionRS updateTestItem(ReportPortalUser.ProjectDetails proj
 		Optional<StatusEnum> statusEnum = StatusEnum.fromValue(rq.getStatus());
 
 		if (statusEnum.isPresent()) {
-			testItem.getItemResults().setIssue(null);
-			testItem.getItemResults().setStatus(statusEnum.get());
+			changeStatus(testItem, statusEnum.get(), user, projectDetails);
```

Comment:
let's move the logic of status changing into different handler
for example ItemStatusChangeHandler